autoload/intero/process.vim

```diff
@@ -278,16 +293,32 @@ function! s:on_stdout(jobid, lines, event) abort
     endfor
 endfunction
 
-function! s:new_response(cmd, response) abort
-    let l:initial_compile = 0
+function! s:parse_ghci_version(output) abort
+    for l in a:output
+        " Try parsing regular GHCi version.
+        let matches = matchlist(l, 'GHCi, version \(\d*\)\.\(\d*\).\(\d*\):')
+        if !empty(matches)
+            let g:intero_ghci_version = [matches[1] + 0, matches[2] + 0, matches[3] + 0]
+        else
+            " Fallback to parsing Intero-style version.
+            let matches = matchlist(l, '(GHC \(\d*\)\.\(\d*\).\(\d*\))')
+            if !empty(matches)
+                let g:intero_ghci_version = [matches[1] + 0, matches[2] + 0, matches[3] + 0]
+            endif
+        endif
+    endfor
+endfunction
```

Comment:
I would prefer this function return the value rather than assigning it out, if only to make (eventually?) testing it easier.

Reply:
Good point, changing that!